### PR TITLE
verbose mode and debug logfile

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

File: ds-collector/ds-collector [hunk 1]
```diff
@@ -1090,7 +1086,17 @@ then
...
-if [ $testMode ]; then
+
+if [ $debug ] ; then
+  set -x
+fi
+if [ $verbose ] ; then
+  bastion_checks
+  logFile="${baseDir}/ds-collector-$(date +'%Y-%m-%d-%H-%M-%s').log"
+  echo "debugging to ${logFile}"
+  ( exec $0 ${@/-v/-x} )  2>&1  | sed "s/${jmxPassword:-'\*\*\*\*'}/****/g" | sed "s/${cqlshPassword:-'\*\*\*\*'}/****/g" | tee "${logFile}"
```
Comment (michaelsembwever) [inline ds-collector/ds-collector:1097]: this is the crux of the new verbose/logging patch

Comment (AndrewHogg) [reply ds-collector/ds-collector:1097]: We need the branch name fix overlaying to be able to test this - and I don't see any of the make files adjusted to use the debug / verbose

Comment (michaelsembwever) [reply ds-collector/ds-collector:1097]: i will add a test that uses it…

File: ds-collector/ds-collector [hunk 2]
```diff
@@ -125,47 +124,49 @@ list_cassandra_nodes() {
...
     $timeout_command nc -zv $(hostname) $jmxPort > /dev/null 2>&1
-    if [ $? == 0 ]; then
+    if [ $? = 0 ]; then
       nodetoolHost="-h $(hostname)"
       echo "Using $(hostname) to connect to JMX..."
     fi
   fi
-
-  cmd="${nodetoolCmd} $nodetoolHost -p $jmxPort ${jmxOpts} $nodetoolCredentials status | grep UN | tr -s ' ' | cut -d' ' -f2"
+  cmd="set -o pipefail ; ${nodetoolCmd} $nodetoolHost -p $jmxPort ${jmxOpts} $nodetoolCredentials status | grep UN | tr -s ' ' | cut -d' ' -f2"
```
[the remark this answers is not in the document]
Comment (michaelsembwever) [reply ds-collector/ds-collector:137]: i'm hesitant to change too much. the script has been developed over many years. let's come back to this. note, putting it into the ssh command is always going to be separately necessary

File: ds-collector/ds-collector [hunk 3]
```diff
@@ -739,6 +740,7 @@ halp() {
     -d                      Run script on a single node only, disabling auto discovery when the -n
                               option specifies a hostname or IP address. Can only be used with
                               Test Mode and Execute Mode.
+    -v                      Verbose mode. Also captures console stdout to a log file under ${baseDir}
```
[the remark this answers is not in the document]
Comment (michaelsembwever) [reply ds-collector/ds-collector:743]: intentionally kept internal.

File: ds-collector/ds-collector [hunk 4]
```diff
@@ -1090,7 +1086,17 @@ then
   jmxPassword=$(escape_special_chars "$jmxPassword")
 fi
 
-if [ $testMode ]; then
+
+if [ $debug ] ; then
+  set -x
```
[the remark this answers is not in the document]
Comment (michaelsembwever) [reply ds-collector/ds-collector:1091]: interesting.

previously we were only uncommenting `set -x` on the bastion's script, so this replaces that use-case. 
let's spin this off as a separate ticket/pr (before it needs careful checking about not leaking credentials)

File: Makefile [hunk 5]
```diff
@@ -87,17 +88,17 @@ ifdef COLLECTOR_SECRETSMANAGER_KEY
 ifdef COLLECTOR_SECRETSMANAGER_SECRET
 	@(command -v aws >/dev/null 2>&1) || { echo >&2 "aws needs to be installed"; exit 1; }
 	@(command -v openssl >/dev/null 2>&1) || { echo >&2 "openssl needs to be installed"; exit 1; }
-	$(eval KEY_FILE_NAME := $(shell echo $${ISSUE}_secret.key))
+	$(eval KEY_FILE_NAME := $(shell echo $$(subst /,-,$(ISSUE))_secret.key))
```
Comment (michaelsembwever) [inline Makefile:91]: i need to test this.

File: ds-collector/ds-collector [hunk 6]
```diff
@@ -427,7 +430,7 @@ s3_push_artifact() {
 
 create_complete_marker() {
   artifact_count="${#artifacts[@]}"
-  node_count="${#cassandraNodes[@]}"
+  node_count="$(echo $cassandraNodes | wc -w)"
```
Comment (michaelsembwever) [inline ds-collector/ds-collector:433]: adding the tests smoke out this failure (`$cassandraNodes` is not an array)

so not strictly part of this PR, but included anyway

**Comment by AndrewHogg** [comment]
Approved

**Pull request merged**